pkg/apis/extensions/v1beta1/defaults.go

```diff
@@ -133,4 +133,14 @@ func SetDefaults_NetworkPolicy(obj *extensionsv1beta1.NetworkPolicy) {
 			}
 		}
 	}
+
+	if obj.Spec.PolicyTypes == nil || len(obj.Spec.PolicyTypes) == 0 {
+		// Default for undefined PolicyTypes with defined Egress rules
+		if obj.Spec.Egress != nil || len(obj.Spec.Egress) != 0 {
+			obj.Spec.PolicyTypes = []extensionsv1beta1.PolicyType{extensionsv1beta1.PolicyTypeIngress, extensionsv1beta1.PolicyTypeEgress}
```

Comment:
So I lost track of what the final decision was. @thockin, is this correct? A policy containing only Egress rules is assumed to also mean default-deny ingress, unless it explicitly contains `policyTypes: [ "Egress" ]` ?

Reply:
I similarly lost track.. Here's what we agreed originally: https://github.com/kubernetes/kubernetes/pull/51808#discussion_r136629478

Reply:
@caseydavenport If we don't assume Ingress, then any policy that is a default-deny-ingress today would change meaning by adding an egress block.  That's not compatible.

Reply:
Right, that makes sense. I like that structuring of it as well.